job-server/src/spark.jobserver/auth/LdapGroupRealm.scala

```diff
@@ -0,0 +1,126 @@
+package spark.jobserver.auth
+
+import org.apache.shiro.realm.ldap.JndiLdapRealm
+import org.apache.shiro.realm.ldap.LdapContextFactory
+import org.apache.shiro.realm.ldap.LdapUtils
+import org.apache.shiro.subject.PrincipalCollection
+import org.apache.shiro.authz._
+import javax.naming.ldap.LdapContext
+import javax.naming.directory._
+import javax.naming.NamingEnumeration
+import org.slf4j.LoggerFactory
+import com.typesafe.config.{ Config, ConfigFactory }
+
+/**
+ * LDAP realm implementation that retrieves group information from LDAP and matches
+ * the 'member' attribute values of each group against the given user.
+ *
+ * @note not all LDAP installations use the member property.... we might have to add
+ *   memberOf matches as well and others
+ *
+ * @author dwk (basics in Java stem from different sources by various authors from stackoverflow and such)
+ */
+class LdapGroupRealm extends JndiLdapRealm {
+
+  private val logger = LoggerFactory.getLogger(getClass)

```

Comment:
@zeitos In my newer projects I use `com.typesafe.scalalogging`.  It has traits that does the above.  We should consider adopting it.  The ActorStack stuff we can revisit as Kamon.io and others have implemented really nice Actor monitoring/debugging stuff.